cub/cub/block/block_load.cuh

```diff
@@ -810,621 +795,273 @@ template <typename InputT,
           int LEGACY_PTX_ARCH          = 0>
 class BlockLoad
 {
-private:
-  /// Constants
-  enum
-  {
-    /// The thread block size in threads
-    BLOCK_THREADS = BLOCK_DIM_X * BLOCK_DIM_Y * BLOCK_DIM_Z,
-  };
+  static constexpr int BLOCK_THREADS = BLOCK_DIM_X * BLOCK_DIM_Y * BLOCK_DIM_Z; // total threads in the block
 
-  /// Load helper
   template <BlockLoadAlgorithm _POLICY, int DUMMY>
-  struct LoadInternal;
+  struct LoadInternal; // helper to dispatch the load algorithm
 
-  /**
-   * BLOCK_LOAD_DIRECT specialization of load helper
-   */
   template <int DUMMY>
   struct LoadInternal<BLOCK_LOAD_DIRECT, DUMMY>
   {
-    /// Shared memory storage layout type
     using TempStorage = NullType;
```

Comment:
Starting from here, I deleted all the documentation blocks of the `LoadInternal` specializations, because they are `private` inside of `BlockLoad` and named `Internal`. Furthermore, their behavior is amply documented in the `BlockLoadAlgorithm` enumeration. Also, the comments were highly redundant and were at least one time wrong.